### ISSUE TITLE
Proposal: wrap encoding settings up in the Codec.

### ISSUE THREAD

**Issue by bsutton** [dooboolab/flutter_sound#337]
When starting a recording the following settings are passed:

 AndroidEncoder androidEncoder = AndroidEncoder.aacCodec,
    AndroidAudioSource androidAudioSource = AndroidAudioSource.mic,
    AndroidOutputFormat androidOutputFormat = AndroidOutputFormat.defaultFormat,
    IosQuality iosQuality = IosQuality.low,
I'm trying to work out which of these can be abstracted away or buried.

Quality - can we/should we make this an attribute of the codec?

AndroidAudioSource.mic - I've look at the set of option. None of them seem particularly relevant for what we are trying to build.
Do our users want access to recording from these sources?

Same with this one:
AndroidOutputFormat

 AndroidEncoder.aacCodec - is this not a duplicate of what Codec does?
I don't understand how this interacts with the Codec class.
If feels like this should be buried in the isEncoderSupported method?

**Comment by Larpoux** [comment]
Yes. In the future V5 I removed those OS specific parameters.
- `AndroidEncoder` is a legacy/obsolete parameter which was used before I introduced the `Codec` parameter (`Codec` is OS independant).
- `AndroidOutputFormat` is also a legacy parameter. Actually, Flutter Sound uses the Codec parameter for specify bot the Encoder and the File format. I think that it is a good choice.
- I suggest to keep a `Quality` parameter, but as an Enum, not Hexa values specific to iOS. If one day we implement Record to MP3, this parameter will be useful both on Android and iOS
- I have not looked carefully to the Audio Source parameter, but we have several "customers!" who actually us it, (and complain that this does not work correctly !)
